File: task/src/main/java/org/kharitonov/springWeb/controllers/TaskController.java
package org.kharitonov.springWeb.controllers;

import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.kharitonov.springWeb.dto.TaskDTO;
import org.kharitonov.springWeb.dto.factories.TaskDTOFactory;
import org.kharitonov.springWeb.entites.TaskEntity;
import org.kharitonov.springWeb.services.TaskService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/task")
public class TaskController {

    private final TaskService taskService;
    private final ObjectMapper objectMapper;

    @Autowired
    public TaskController(TaskService taskService, ObjectMapper objectMapper) {
        this.taskService = taskService;
        this.objectMapper = objectMapper;
    }

    @GetMapping("/tasks")
    public String getAll() throws JsonProcessingException {
        List<TaskDTO> taskDTOS = taskService.findAll()
                .stream()
                .map(taskEntity -> new TaskDTOFactory().toTaskDTO(taskEntity))
                .toList();
        return objectMapper.writeValueAsString(taskDTOS);
    }

    @PostMapping
    public String createTask(@RequestBody Map<String,String> bodyParams) throws JsonProcessingException {
        TaskEntity taskEntity = new TaskEntity.Builder()
                .name(bodyParams.get("name"))
                .description(bodyParams.get("description"))
                .build();
        taskService.save(taskEntity);
        return objectMapper.writeValueAsString(new TaskDTOFactory().toTaskDTO(taskEntity));
    }

    @PatchMapping("/{id}")
    public String updateTask(@PathVariable("id") int id, @RequestBody Map<String, String> bodyParams) throws JsonProcessingException {
        TaskEntity taskEntityFromDB = taskService.findById(id);
        TaskEntity taskEntity = new TaskEntity.Builder()
                .name(bodyParams.get("name"))
                .description(bodyParams.get("description"))
                .build();
        TaskEntity entity = taskService.copyProperties(taskEntityFromDB, taskEntity);
        taskService.save(entity);
        TaskDTO taskDTO = new TaskDTOFactory().toTaskDTO(entity);
        return objectMapper.writeValueAsString(taskDTO);
    }

    @DeleteMapping("/{id}")
    public void deleteTask(@PathVariable("id") int id) {
        taskService.deleteById(id);
    }
}

File: laba_chetyre/src/main/kotlin/by/bsut/labachetyre/controllers/TaskController.kt
package by.bsut.labachetyre.controllers

import com.fasterxml.jackson.core.JsonProcessingException
import com.fasterxml.jackson.databind.ObjectMapper
import by.bsut.labachetyre.dto.TaskDTO
import by.bsut.labachetyre.dto.factories.TaskDTOFactory
import by.bsut.labachetyre.entites.TaskEntity
import by.bsut.labachetyre.services.TaskService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/task")
class TaskController @Autowired constructor(

    private val taskService: TaskService,
    private val objectMapper: ObjectMapper
) {

    @get:Throws(JsonProcessingException::class)
    @get:GetMapping("/tasks")
    val all: String get() = objectMapper.writeValueAsString(
        taskService.findAll().stream().map { taskEntity -> TaskDTOFactory().toTaskDTO(taskEntity) }.toList()
    )

    @PostMapping("/{id}")
    @Throws(JsonProcessingException::class)
    fun createTask(@RequestBody bodyParams: Map<String?, String?>): String {
        val taskEntity: TaskEntity = getTaskEntity()
        taskService.save(taskEntity)
        return objectMapper.writeValueAsString(TaskDTOFactory().toTaskDTO(taskEntity))
    }

    @PatchMapping("/{id}")
    @Throws(JsonProcessingException::class)
    fun updateTask(@PathVariable("id") id: Int, @RequestBody bodyParams: Map<String?, String?>): String {
        val entity: TaskEntity = taskService.copyProperties(taskService.findById(id), getTaskEntity())
        taskService.save(entity)
        return objectMapper.writeValueAsString(TaskDTOFactory().toTaskDTO(entity))
    }

    @DeleteMapping("/{id}")
    fun deleteTask(@PathVariable("id") id: Int) = taskService.deleteById(id)

    private fun getTaskEntity() = Builder().name(bodyParams["name"]).description(bodyParams["description"]).build()
}